ekncontent/ekncontent/eknc-domain.c

```diff
@@ -820,6 +908,16 @@ on_xapian_query_response (GObject *source,
       return;
     }
 
+  state->models = g_ptr_array_new_full (state->total_models, maybe_unref_object);
+  g_ptr_array_set_size (state->models, state->total_models);
```

Comment:
Oh, I see...  I guess you want the size to match `state->total_models` because of how you use `g_ptr_array_index()` in the callback, so this makes sense.